sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -105,6 +105,66 @@ trait InvokeLike extends Expression with NonSQLExpression {
   }
 }
 
+/**
+ * Common trait for [[DecodeUsingSerializer]] and [[EncodeUsingSerializer]]
+ */
+trait BaseSerializer {
```

Comment:
`BaseSerializer` is a bit generic. Can we come up with a better name? We could name it `UsingSerializer` or `SerializerSupport` (I am not super excited about either, so we could also leave it as it is).

Reply:
I can't think of any better name honestly.... I'd go for `SerializerSupport`, thanks.